app/services/lms_sync_service.py

```diff
@@ -177,6 +191,11 @@ async def upsync_grade(
         comments: str | None = None,
     ):
         user_pid = await self.canvas_service.get_pid_from_onyen(submission.student.onyen)
+        
+        # If this course runs on a 2U Digital Campus instance, append ":UNC" to the PID
+        if "digitalcampus" in settings.CANVAS_API_URL:
```

Comment:
I think you should put this line before all of the for loops you've added

Reply:
If you disagree though I don't think it's a big deal

Reply:
yeah this check could be added, though the flow would be the same. since it's a temporary addition to this file I am not big on adding more checks, besides we'll remove it in favour of a more robust solution the first chance we get